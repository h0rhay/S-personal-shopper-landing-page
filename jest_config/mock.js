export const mockTheme = {
  themeType: 'dark',
  breakpoints: {
    small: '(max-width: 812px)',
    medium: '(min-width: 813px)',
    large: '(min-width: 960px)',
    deviceSwitchSm: '(max-width: 896px)',
    deviceSwitchLg: '(min-width: 897px)',
  },
  typography: {
    mobile: {
      headerOne: {
        fontSize: '3rem',
      },
      headerTwo: {
        fontSize: '1.75rem',
      },
      displayHeading: {
        fontSize: 'calc(1.5rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
    desktop: {
      headerOne: {
        fontSize: '4rem',
      },
      headerTwo: {
        fontSize: '2rem',
      },
      displayHeading: {
        fontSize: 'calc(4rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
  },
  text: '#ffffff',
  primary: '#212121',
  primaryAccent: '#2E2E2E',
  accent: '#ffe256',
  shadow: 'rgba(0, 0, 0, 0.4)',
};
