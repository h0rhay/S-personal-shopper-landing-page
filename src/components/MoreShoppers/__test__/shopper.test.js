import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../../../jest_config/mock';
import Shopper from '../shopper';

jest.mock('../../../hooks/useElementOnScreen', () => jest.fn().mockReturnValue([{ current: 'test' }, true]));
jest.mock('gatsby-link', () => <div>Link</div>);

let container;
const props = {
  index: 0,
  shopper: {
    id: 'id',
    frontmatter: {
      id: 'testId',
      firstName: 'testFirstName',
      lastName: 'testLastName',
      title: 'Test Title',
      slug: 'test',
      expertise: 'Test Expertise',
      profileImage: {
        relativePath: 'images/test.jpg',
        childImageSharp: {
          fluid: {
            srcSet:
              '/static/e97b0bd1cdb49dc6febfe800eec9b70b/fd013/arron-dickinson.jpg 200w, /static/e97b0bd1cdb49dc6febfe800eec9b70b/25252/arron-dickinson.jpg 400w, /static/e97b0bd1cdb49dc6febfe800eec9b70b/c3638/arron-dickinson.jpg 600w',
            src: '/static/b991240d56e4b1cff4fc035662ba8b36/6e63d/connie-profile-img.jpg',
            alt: 'Connie Jones',
          },
        },
      },
    },
  },
};

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <Shopper {...props} />
    </ThemeProvider>
  );

describe('Shopper', () => {
  beforeEach(() => {
    container = renderComponent({ theme: mockTheme }).container;
  });
  it('should match screenshot', () => {
    expect(container).toMatchSnapshot();
  });
});
