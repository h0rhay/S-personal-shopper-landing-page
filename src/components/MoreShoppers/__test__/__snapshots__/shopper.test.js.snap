// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shopper should match screenshot 1`] = `
<div>
  <div
    class="sc-hKwDye fkoMWK"
  >
    <div
      class="sc-bdvvtL bdzgBJ"
      style="animation-duration: 400ms; animation-delay: 200ms;"
    >
      <a
        class="sc-eCImPb bfyOOk"
        href="/test"
      >
        <div
          class="sc-gsDKAQ bwQhCI sc-dkPtRN iAbveQ"
          dir="shopper"
        >
          <img
            alt="testFirstName testLastName"
            class="sc-jRQBWg kQUKKv"
            src="/static/b991240d56e4b1cff4fc035662ba8b36/6e63d/connie-profile-img.jpg"
            srcset="/static/e97b0bd1cdb49dc6febfe800eec9b70b/fd013/arron-dickinson.jpg 200w, /static/e97b0bd1cdb49dc6febfe800eec9b70b/25252/arron-dickinson.jpg 400w, /static/e97b0bd1cdb49dc6febfe800eec9b70b/c3638/arron-dickinson.jpg 600w"
          />
        </div>
        <p
          class="sc-gKclnd dhKRsh"
        >
          Test Expertise
        </p>
        <h2>
          testFirstName
        </h2>
        <h2
          class="sc-iCfMLu cxkPBJ"
        >
          testLastName
        </h2>
        <div
          class="sc-furwcr ghnaZL"
        >
          Test Title
        </div>
      </a>
    </div>
  </div>
</div>
`;
